src/lib.rs

```diff
@@ -622,6 +630,20 @@ impl Config {
     pub fn set_disable_active_migration(&mut self, v: bool) {
         self.local_transport_params.disable_active_migration = v;
     }
+
+    /// Sets the congestion control algorithm used (string).
+    ///
+    /// The default value is 'reno'.
+    pub fn set_cc_algorithm(&mut self, name: &str) {
```

Comment:
We should just make an enum with all the possible CC algorithms, rather than bypass the type system by using strings. This will make it much easier to validate that the user passed a valid value (as the rust compiler will do that), as well as make it easier to document the allowed values.

Reply:
Currently this API is similar to setsockopt(...TCP_CONGESTION...) so string is used. I was also thinking about enum... let me update.

Reply:
Changed to enum cc::Algorithm.

However, for examples, I added config.set_cc_algorithm_name() API which accepts a string. Otherwise when command line is parsed there need to be something which translates a string to cc::Algorithm enum. e.g. "reno" -> cc::Algorithm.

I added cc::lookup_cc_algorithm() method for that, but `cc` module itself is not visible to the application, so added a new config API. Any better idea is welcome.

Reply:
My point was that we should also make the enum public, so the config API can just take a value from that and then you don't need to convert the string.

You can export the `cc::Algorithm` enum in `lib.rs` like:
```rust
pub use crate::cc::Algorithm CongestionControlAlgorithm;
```

Reply:
Good tip! Added 'as' in between.